{
  "id": "uno-ooo-parser",
  "version": "0.2.15",
  "libre_office_ver": "7.4",
  "name": "Exception",
  "type": "exception",
  "namespace": "com.sun.star.uno",
  "parser_args": {
    "sort": false,
    "long_names": true,
    "remove_parent_inherited": true
  },
  "writer_args": {
    "include_desc": true
  },
  "data": {
    "allow_db": true,
    "from_imports": [],
    "from_imports_typing": [
      [
        ".x_interface",
        "XInterface",
        "XInterface_8f010a43"
      ]
    ],
    "extends_map": {},
    "quote": [
      "XInterface_8f010a43"
    ],
    "typings": [],
    "requires_typing": true,
    "full_imports": {
      "general": [],
      "typing": [
        "com.sun.star.uno.XInterface"
      ]
    },
    "imports": [],
    "name": "Exception",
    "namespace": "com.sun.star.uno",
    "extends": [
      "Exception"
    ],
    "desc": [
      "the base of all UNO exceptions",
      "",
      "All exceptions defined in UNO idl should derive from this exception."
    ],
    "url": "https://api.libreoffice.org/docs/idl/ref/exceptioncom_1_1sun_1_1star_1_1uno_1_1Exception.html",
    "items": {
      "properties": [
        {
          "name": "Message",
          "returns": "str",
          "origtype": null,
          "origin": "string",
          "desc": [
            "gives a detailed description of the reason, why the exception was thrown.",
            "",
            "The description should be as detailed as possible."
          ],
          "raises_get": "",
          "raises_set": ""
        },
        {
          "name": "Context",
          "returns": "XInterface_8f010a43",
          "origtype": "com.sun.star.uno.XInterface",
          "origin": "com.sun.star.uno.XInterface",
          "desc": [
            "should contain a reference to the original, which raised the exception.",
            "",
            "May be NULL."
          ],
          "raises_get": "",
          "raises_set": ""
        }
      ]
    }
  }
}